File: mealPlan.py
'''
Script to generate a weekly meal plan with random selections from predefined lists of breakfasts, lunches, and dinners.

The script prompts the user to choose between having one or two meals per day for variety. It then randomly selects meals 
from the predefined lists and displays the weekly meal plan. If two meals per day are chosen, it ensures that each meal is 
unique for variety and avoids repetition of the last item in each list. After displaying the meal plan, it allows the user to 
opt for a different selection or end the program.

Usage:
    - Run the script and respond to prompts with 'Y' or 'N' to generate a meal plan with desired variety.

Predefined Lists:
    - `breakfasts`: List of breakfast options.
    - `lunches`: List of lunch options.
    - `dinners`: List of dinner options.

Example:
    $ python meal_plan_generator.py
    Would you like some variety in your week? (Y or N) Y
    Okay, this week you're gonna have scrambled eggs for breakfast, pesto stuffed shells for lunch, and jerk chicken with 
    pineapple black bean salsa for dinner.
    
    You will also have bacon potato frittata for breakfast, chicken & dumplins for lunch, and sesame garlic ramen stir fry for 
    dinner.

    Would you rather have something different? (Y or N) N

    Bon Appetit!
'''
import random

breakfasts = [
  "sweet potato hash",
  "strawberry overnight oatmeal",
  "egg muffin","cottage cheese breakfast bowls", 
  "country breakfast bowls", "microwave breakfast scrambles",
  "smoothie packs", "spicy breakfast strata", 
  "crock pot sausage casserole", 'bagel sandwiches',
  'jalapeno poppy seed muffins', 'pizza eggs',
  'breakfast enchiladas', 'eggs benedict (df2)', 'egg cups',
  'egg boats (df2)', 'bacon potato frittata (ww)',
  'breakfast quinoa fried rice (ww)',
  'breakfast quesadillas (ww)'
] 

lunches = [
  "mason jar salad", 'taco salad (df2)' 
  "steak cobb salad",
  "sandwiches","skillet meatballs & marina",
  "cilantro lime chicken", "pizza roll ups",
  "loaded sweet potato with cilantro lime crema",
  "sweet potato tacos", "pesto stuffed shells",
  'frito chili pie', 'taco soup', 'chicken & spinach flautas',
  'veggie tortellini soup (pw)',
] 

dinners = [
  'baked chicken fingers (df2)', 'hamburger helper (df2)',
  'ranch chicken tacos', 'tex-mex butternut squash (pw)',
  'pesto chicken', 'spinach lasagna rolls (df2)', 
  'chip crusted pork chops (df2)', 'steak with chimichurri (pw)',
  'tuscan chicken sheet pan (pw)', 'chicken bacon ranch pasta',
  "one pan huevos rancheros", "chicken ramen stir fry",
  "sesame garlic ramen stir fry",
  "chicken parm pasta bake", "honey garlic chicken fried rice",
  "chicken with creamy cilantro lime sauce", "tex mex stuffed zucchini",
  "beef & broccoli ramen stir fry",
  "jerk chicken with pineapple black bean salsa", 'paninis',
  'lasagne (df2)', 'gumbo (df2)', 'spaghetti with italian sausage',
  'spaghetti (pw)','beef stew', 'subs'
  'tortilla pizza', 'carne asada taco bowls',
  'crock pot red beans & rice', 'chicken parm wraps', 
  'mississippi roast', 'carne asada quesadillas',
  'sweet tea fried chicken (df2)', 
  'chicken with clilantro lime rice (df2)',
  'chicken & dumplins (df2)', 'sheet pan pork loin and veggies (df2)',
  'chicken pot pie (df2)', 'beef enchiladas (df2)',
  'skillet pizza (df2)', "cap'n crunch chicken strips (pw)",
  'mango chile chicken (pw)', 'smothered pork chops (pw)',
  'chicken piccata (pw)', 'taco quesadillas (pw)',
  'turkey taco skillet (pw)',
  'spaghetti carbonara with grilled chicken (pw)', 'dinner nachos (pw)',
  'pepperoni chicken (pw)', 'parmesan panko chicken (pw)',
  'chicken + 40 cloves of garlic (pw)', 'sausage + root veggies sheet pan (pw)',
  'pizza margherita (ww)', 'greek chicken (ww)', 'easy chicken parm (ww)',
  'bbq chicken skillet pizza (ww)', 'pizza casserole', 
  'country quiche (ty)', 'tennessee jambalya (ty)', 'pumpkin chili',
  'slow cooker pork loin (ty)', 'squash casserole (ty)',
  "baked mac 'n cheese (df2)", 'quick and easy ravioli soup (ww)',
  'easy make at home chinese chicken (ww)', 'southwestern turkey stew (ww)',
  'classic fettuccine alfredo (ww)', 'nerd fitness chicken',
] 

print('===================================================')
print('               Meal Plan Generator')
print('===================================================')

number_of_meals = input("\nWould you like some variety in your week? (Y or N) ")
if number_of_meals.lower() == "y":
  number_of_meals = 2
else:
  number_of_meals = 1

'''
For the following 3 sets of nested `if` statements, the goal is to ensure that both meal options are unique. If `random.choice()`
selects 2 identical meals, and that meal is the last element in the list, the second meal will be set to the 
second-to-last element in the list, else the second meal will be set to the last element in the list
'''

while True:
  breakfast = random.choice(breakfasts)
  lunch = random.choice(lunches)
  dinner = random.choice(dinners)

  if number_of_meals == 2:
    breakfast1 = breakfast
    breakfast2 = breakfast
    lunch1 = lunch
    lunch2 = lunch 
    dinner1 = dinner
    dinner2 = dinner
    
    if breakfast1 == breakfast2 and breakfast1 == breakfasts[-1]:
      breakfast2 = breakfasts[-2]
    elif breakfast1 == breakfast2:
      breakfast2 = breakfasts[-1]
   

    if lunch1 == lunch2 and lunch1 == lunches[-1]:
      lunch2 = lunches[-2]
    elif lunch1 == lunch2:
      lunch2 = lunches[-1]

    if dinner1 == dinner2 and dinner1 == dinners[-1]:
      dinner2 = dinners[-2]
    elif dinner1 == dinner2:
      dinner2 = dinners[-1]
   
    print(f"\nOkay, this week you're gonna have {breakfast1} for breakfast, {lunch1} for lunch, and {dinner1} for dinner.") 
    print(f"\nYou will also have {breakfast2} for breakfast, {lunch2} for lunch, and {dinner2} for dinner.") 
  
  else:
    print(f"\nOkay, this week you're gonna have {breakfast} for breakfast, {lunch} for lunch and {dinner} for dinner.") 


  response = input("\nWould you rather have something different? (Y or N) ")
  if response.lower() == "n":
    print("\n\nBon Appetit!\n\n") 
    break

File: mealPlan.js
// Meal lists now include the meal name and a corresponding link to the recipe page
const breakfasts = [
  { name: "one pan huevos rancheros", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/huevosRancheros.pdf" },
  { name: "crockpot sausage & cheese casserole", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/sausageCheeseBreakfastCasserole.pdf" },
  { name: "Spicy Breakfast Strata", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/spicyBreakfastStrata.pdf" },
  { name: "PB & J Cottage Cheese Breakfast Bowl", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Peanut%20Butter%20and%20Jelly%20Cottage%20Cheese%20Breakfast%20Bowl%20-%20Budget%20Bytes.pdf" },
  { name: "Microwave Breakfast Scrambles", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Make%20Ahead%20Microwave%20Breakfast%20Scrambles%20-%20Budget%20Bytes.pdf" },
  { name: "Country Breakfast Bowls", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Country%20Breakfast%20Bowls%20-%20Budget%20Bytes.pdf" },
  { name: "Butternut Squash & Brussels Sprout Breakfast Quesadillas", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Butternut%20and%20Brussels%20Sprout%20Breakfast%20Quesadillas%20-%20Domesticate%20ME.pdf" },
  { name: "Shakshuka", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Shakshuka%20-%20Dessert%20for%20Two.pdf" },
  // Add more breakfasts with corresponding links
];

const lunches = [
  { name: "chicken ramen stir-fry", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/chickenRamenStirFry.pdf" },
  { name: "honey garlic chicken fried rice", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/honeyGarlicChickenFriedRice.pdf" },
  // Add more lunches with corresponding links
];

const dinners = [
  { name: "beef broccoli ramen stir-fry", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/beefBroccoliRamenStirFry.pdf" },
  { name: "chicken with creamy cilantro lime sauce", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/chickenCreamyCilantroLimeSauce.pdf" },
  { name: "chicken parmesan pasta bake", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/chickenParmesanPastaBake.pdf" },
  { name: "Sesame Garlic Ramen Stir-Fry", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/sesameGarlicRamenStirFry.pdf" },
  { name: "Chicken with Creamy Cilantro Lime Sauce", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/chickenCreamyCilantroLimeSauce.pdf" },
  { name: "Tex Mex Stuffed Zucchini", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/texMexStuffedZucchini.pdf" },
  { name: "Pesto Stuffed Shells", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Pesto%20Stuffed%20Shells%20Meal%20Prep%20-%20Budge.md" },
  { name: "Skillet Meatballs with Marina & Broccoli", link: "hhttps://github.com/jeremyraby/meal_plan_generator/blob/main/Sunday%20Meal%20Prep_Skillet%20Meatballs%20and%20Marinara.md" },
  { name: "Sweet Potato Tacos", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Sweet%20Potato%20Taco%20Meal%20Prep%20-%20Budget%20B.md" },
  { name: "Pizza Roll Ups", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/The%20Pizza%20Roll%20Up%20Lunch%20Box%20-%20Budget%20Bytes.pdf" },
  { name: "Sweet Potatoes with Lime Crema", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Loaded%20Sweet%20Potatoes%20with%20Lime%20Crema%20-%20Budget%20Bytes.pdf" },
  { name: "Ground Beef Tacos", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Ground%20Beef%20for%20Tacos%20-%20Dessert%20for%20Two.pdf" },
  { name: "Graham Cracker Fried Chicken", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Graham%20Cracker%20Fried%20Chicken%20-%20Dessert%20for%20Two.pdf" },
  { name: "Skillet Chicken Pot Pie", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Chicken%20Pot%20Pie%20Skillet%20Recipe%20-%20Dessert%20for%20Two.pdf" },
  { name: "Broccoli Rice Casserole with Chicken", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Broccoli%20Rice%20Casserole%20with%20Chicken%20-%20Dessert%20for%20Two.pdf" },
  { name: "Baked Breaded Chicken Strips (Freezer)", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Baked%20Breaded%20Chicken%20Strips%20-%20Dessert%20for%20Two.pdf" },
  { name: "PB Steak Tacos", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Peanut%20Butter%20Steak%20Tacos%20(with%20sriracha!)%20-%20Dessert%20for%20Two.pdf" },
  { name: "Pork Meatballs", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Pork%20Meatballs%20-%20Dessert%20for%20Two.pdf" },
  { name: "Skillet Lasagna", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Skillet%20Lasagna%20-%20Dessert%20for%20Two.pdf" },
  { name: "Stuffed Bell Peppers", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/Stuffed%20Peppers%20for%20Two%20-%20Dessert%20for%20Two.pdf" },
  { name: "White Chicken & Green Chile Enchiladas", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/White%20Chicken%20Enchilada%20Recipe%20-%20Dessert%20for%20Two.pdf" },
  { name: "Poblano Pepper Soup", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/poblano_pepper_soup.md" },
  { name: "Italian Cream Cheese Chicken & Creamy Chicken Soup", link: "https://github.com/jeremyraby/meal_plan_generator/blob/main/italianCreamCheeseChicken_creamyChickenSoup.pdf" },
  // Add more dinners with corresponding links
];

// Helper function to get a random item from a list
function getRandomItem(list) {
  return list[Math.floor(Math.random() * list.length)];
}

// Event listeners for buttons
document.getElementById('yesVariety').addEventListener('click', () => generateMealPlan(2));
document.getElementById('noVariety').addEventListener('click', () => generateMealPlan(1));

// Generate meal plan function
function generateMealPlan(numberOfMeals) {
  let mealOutput = '';
  
  let breakfast1 = getRandomItem(breakfasts);
  let lunch1 = getRandomItem(lunches);
  let dinner1 = getRandomItem(dinners);

  if (numberOfMeals === 2) {
    // Ensure unique selections for second meal if variety is chosen
    let breakfast2 = getRandomItem(breakfasts);
    let lunch2 = getRandomItem(lunches);
    let dinner2 = getRandomItem(dinners);

    if (breakfast1.name === breakfast2.name) breakfast2 = breakfasts[breakfasts.length - 1];
    if (lunch1.name === lunch2.name) lunch2 = lunches[lunches.length - 1];
    if (dinner1.name === dinner2.name) dinner2 = dinners[dinners.length - 1];
    if (dinner1.name === 'Italian Cream Cheese Chicken & Creamy Chicken Soup') dinner2 = 'Creamy Chicken Soup';
    if (dinner2.name === 'Italian Cream Cheese Chicken & Creamy Chicken Soup') dinner2 = dinners[dinners.length - 1];
    
    mealOutput = `
      <p>Okay, this week you're gonna have <a href="${breakfast1.link}" target="_blank">${breakfast1.name}</a> for breakfast, 
      <a href="${lunch1.link}" target="_blank">${lunch1.name}</a> for lunch, and 
      <a href="${dinner1.link}" target="_blank">${dinner1.name}</a> for dinner.</p>
      
      <p>You will also have <a href="${breakfast2.link}" target="_blank">${breakfast2.name}</a> for breakfast, 
      <a href="${lunch2.link}" target="_blank">${lunch2.name}</a> for lunch, and 
      <a href="${dinner2.link}" target="_blank">${dinner2.name}</a> for dinner.</p>
    `;
  } else {
    mealOutput = `
      <p>Okay, this week you're gonna have <a href="${breakfast1.link}" target="_blank">${breakfast1.name}</a> for breakfast, 
      <a href="${lunch1.link}" target="_blank">${lunch1.name}</a> for lunch, and 
      <a href="${dinner1.link}" target="_blank">${dinner1.name}</a> for dinner.</p>
    `;
  }

  // Display meal plan and show options for new selection
  document.getElementById('mealOutput').innerHTML = mealOutput;
  document.getElementById('mealPlan').style.display = 'block';
}

// Event listener for different meal option
document.getElementById('yesDifferent').addEventListener('click', () => {
  document.getElementById('mealPlan').style.display = 'none';
  generateMealPlan(2); // Default to 2 meals for variety
});

document.getElementById('noDifferent').addEventListener('click', () => {
  document.getElementById('mealPlan').style.display = 'none';
  alert("Bon Appetit!");
});
